tests/v3/test_store/test_stateful_store.py

```diff
@@ -0,0 +1,190 @@
+# Stateful tests for arbitrary Zarr stores.
+
+import asyncio
+
+import hypothesis.strategies as st
+from hypothesis import note
+from hypothesis.stateful import (
+    RuleBasedStateMachine,
+    invariant,
+    precondition,
+    rule,
+)
+
+from zarr.abc.store import Store as StoreABC
+from zarr.buffer import Buffer, default_buffer_prototype
+from zarr.store import LocalStore, MemoryStore, RemoteStore
+
+# from strategies_store import StoreStatefulStrategies, key_ranges
+from zarr.testing.strategies import key_ranges, paths
+
+# zarr spec: https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html
+
+
+class SyncMemStoreWrapper(StoreABC):
+    def __init__(self, store_type):
+        """Store to hold async functions that map to StoreABC abstract methods"""
+        if store_type in ("Memory", "memory"):
+            self.store = MemoryStore(mode="w")  # store_container.store
+        elif store_type in ("Local", "local"):
+            self.store = LocalStore(mode="w")
+        elif store_type in ("Remote", "remote"):
+            self.store = RemoteStore(mode="w")
+        else:
+            raise ValueError(f"Invalid item_type: {store_type}")
```

Comment:
```suggestion
    def __init__(self, store):
        """Store to hold async functions that map to StoreABC abstract methods"""
        self.store = store
```

This let's you do `SyncStoreWrapper(MemoryStore(mode="w"))` or any other store that satisfies `StoreABC`